packages/dds/tree/src/util/rangeMap.ts

```diff
@@ -6,229 +6,239 @@
 import { oob } from "@fluidframework/core-utils/internal";
 
 /**
- * A map keyed on integers allowing reading and writing contiguous ranges of integer keys.
- *
- * TODO: We should avoid the direct exposure of RangeEntry. AB#7414
+ * RangeMap represents a mapping from integers to values of type T or undefined.
+ * The values for a range of consecutive keys can be changed or queried in a single operation.
  */
-export type RangeMap<T> = RangeEntry<T>[];
+export class RangeMap<T> {
+	private readonly entries: RangeEntry<T>[];
 
-export interface RangeEntry<T> {
-	start: number;
-	length: number;
-	value: T;
-}
+	public constructor(initialEntries?: RangeEntry<T>[]) {
```

Comment:
```suggestion
	public constructor(initialEntries?: readonly RangeEntry<T>[]) {
```

Reply:
The internals of each `RangeEntry` are still mutable though. Consider making them immutable.

Reply:
The constructor copies `initialEntries`, so I don't think there's any reason to make the array readonly. I did make the fields readonly.

Reply:
Ah, that's right. Nice!

Reply:
FWIW - making the input readonly does not tighten the contract - it actually loosens it. A read+write array satisfies 'readonly array`. IMO it is best practice to always make the input readonly unless you need to mutate it - so, in this case, readonly would be appropriate.